extensions/battleroyale/standalone.mjs

```diff
@@ -0,0 +1,338 @@
+   */
+  constructor(ID, facing, mapCoord) {
+    this.ID = ID;
+    this.facing = facing;
+    this.mapCoord = mapCoord;
+    this.duration = 0;
+    this.intervalID = undefined;
+    this.mapCoord.x += Facing2Direction[this.facing][0]; // move one
+    this.mapCoord.y += Facing2Direction[this.facing][1];
+  }
+
+  /**
+   * update the bullet coord
+   * @return {undefined}
+   */
+  updateCoord() {
+    this.mapCoord.x += Facing2Direction[this.facing][0];
+    this.mapCoord.y += Facing2Direction[this.facing][1];
+    this.duration += 1;
+  }
+
+  /**
+   * retrun bullet in this coord
+   * @param {MapCoord} coord
+   * @return {Boolean}
+   */
+  atCoord(coord) {
+    return this.mapCoord.x == coord.x &&
+     this.mapCoord.y == coord.y;
+  }
+
+  /**
+   * delete the interval
+   * @return {undefined}
+   */
+  deleteInterval() {
+    if (this.intervalID !== undefined) {
+      clearInterval(this.intervalID);
+    }
+  }
+}
+
+/**
+ * This represents the standalone extension service for this extension.
+ */
+class Standalone {
+  /**
+   * Create the standalone extension service object but does not start it.
+   * @constructor
+   * @param {ExtensionHelper} helper - An extension helper object for servicing
+   * various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+  }
+
+  /**
+   * Initializes the extension.
+   */
+  async initialize() {
+    // get the default arena of every map
+    this.arenaOfMaps = this.helper.gameMap.getOriginalCellSetStartWith('battleroyaleArena');
+    this.obstaclesOfMaps = this.helper.gameMap.getOriginalCellSetStartWith('battleroyaleObstacles');
+
+    this.bullets = new Map(); // key: bullet ID; value: bullet info
+    this.bulletID = 0;
+
+    this.fires = new Set();
+
+    for (const mapName of this.arenaOfMaps.keys()) {
+      await this.helper.broadcastCellSetUpdateToAllUser(
+          'set',
+          mapName,
+          CellSet.fromObject({
+            name: LAYER_BULLET.layerName,
+            priority: LAYER_BULLET.zIndex,
+            cells: Array.from(this.bullets.values()),
+            layers: {[LAYER_BULLET.layerName]: 'BB'},
+            dynamic: true,
+          }),
+      );
+      await this.helper.broadcastCellSetUpdateToAllUser(
+          'set',
+          mapName,
+          CellSet.fromObject({
+            name: LAYER_FIRE.layerName,
+            priority: LAYER_FIRE.zIndex,
+            cells: Array.from(this.fires),
+            layers: {[LAYER_FIRE.layerName]: 'BF'},
+            dynamic: true,
+          }),
+      );
+    }
+
+    this.cooldown = new Set();
+    this.fireCounter = 0;
+    const updateMap = setInterval( async () => {
+      this.fireCounter++;
+      this.fires.clear();
+      for (const mapName of this.arenaOfMaps.keys()) {
+        const needUpdate = this.updateEdge(
+            this.fires, this.arenaOfMaps.get(mapName), this.fireCounter,
+        );
+        if (!needUpdate) {
+          clearInterval(updateMap);
```

Comment:
We might have to design an auto restart mechanism, such that we'll have multiple rounds running one after another, or players may be able to trigger it to start.